packages/web/src/pages/profile-page/components/desktop/ProfileTopTags.tsx

```diff
@@ -20,48 +13,28 @@ const messages = {
 const MOST_USED_TAGS_COUNT = 5
 
 export const ProfileTopTags = () => {
-  const dispatch = useDispatch()
   const user = useProfileParams()
-  const handle = user?.handle
-  const userId = user?.user_id
-
-  const topTagsStatus = useSelector((state: AppState) => {
-    if (!handle) return Status.IDLE
-    return getTopTagsStatus(state, handle)
-  })
+  const { data, status } = useTopTags(user?.user_id)
+  const topTags = data?.slice(0, MOST_USED_TAGS_COUNT)
 
-  const topTags = useSelector((state: AppState) => {
-    if (handle) {
-      return getTopTags(state, handle)?.slice(0, MOST_USED_TAGS_COUNT)
-    }
-  })
+  if (status !== 'success' || !topTags?.length) return null
```

Comment:
we have been preferring the `isPending` `isSuccess` and `isError` booleans returned in the query results instead of matching status manually

```suggestion
  if (isPending || !topTags?.length) return null
```

Reply:
ah nice